core/templates/dev/head/pages/exploration_editor/EditorServices.js

```diff
@@ -1028,6 +1039,29 @@ oppia.factory('explorationStatesService', [
     return {
       init: function(statesBackendDict) {
         _states = StatesObjectFactory.createFromBackendDict(statesBackendDict);
+        // Initialize the solutionValidityService.
+        SolutionValidityService.init();
+        _states.getStateNames().forEach(function(stateName) {
+          var solution = _states.getState(stateName).interaction.solution;
+          if (solution) {
+            AnswerClassificationService.getMatchingClassificationResult(
+              explorationContextService.getExplorationId(),
+              _states.getState(stateName),
+              solution.correctAnswer,
+              true,
+              $injector.get(angularNameService.getNameOfInteractionRulesService(
+                _states.getState(stateName).interaction.id))
+            ).then(function(result) {
+              if (
```

Comment:
Try to avoid branching in your code if possible. Consider instead:

```
    var solutionIsValid = (
      editorContextService.getActiveStateName() !== result.outcome.dest);
    SolutionValidityService.updateValidity(stateName, solutionIsValid);
```